src/workforce-management/Controllers/DepartmentController.cs

```diff
@@ -44,9 +51,88 @@ public DepartmentController(BangazonContext ctx)
-        }
-    }
+        }
+
+
+        /**
+          * Purpose: Provides the Add view
+          * Arguments:
+          *    N/A
+          * Return:
+          *     Redirect to the Department Index if the model is valid, will redirect the user back to the form if it is invalid. 
+          **/
+        [HttpGet]
+        public IActionResult Add()
+        {
+            var model = new SingleDepartment();
+            model.Employees = context.Employee
+                    .OrderBy(e => e.LastName)
+                    .AsEnumerable()
+                    .Where(e => e.EndDate == null)
+                    .Select(li => new SelectListItem
+                    {
+                        Text = li.LastName + " " + li.FirstName,
+                        Value = li.EmployeeId.ToString()
+                    });
+            return View(model);
+        }
+
+        /**
+         * Purpose: Provides the Add view
+         * Arguments:
+         *    This method takes a newdepartment as a paramenter, the new department is the one you would like to post into the database.
+         * Return:
+         *     Redirect to the Department Index if the model is valid, will redirect the user back to the form if it is invalid. 
+         **/
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Add(SingleDepartment singleDepartment)
+        {
+            if (ModelState.IsValid)
+            {
+                Department newDepartment = singleDepartment.NewDepartment;
+                System.Console.WriteLine(newDepartment);
+                System.Console.WriteLine(singleDepartment);
+                System.Console.WriteLine(newDepartment.Employees);
+
+                context.Add(newDepartment);
+                if (singleDepartment.Employees != null)
+                {
+                    Console.WriteLine("number 1");
+                }
+
+                if (singleDepartment.NewDepartment.Employees != null)
+                {
+                    Console.WriteLine("number 2");
+                }
+
+                if (newDepartment.Employees != null)
+                {
+               System.Console.WriteLine("number 3");
+
+                    foreach (Employee employee in newDepartment.Employees)
+                    {
+                        Employee employeeToChange = await context.Employee.SingleAsync(e => e.EmployeeId == employee.EmployeeId);
+                        System.Console.WriteLine(">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>");
+                        employeeToChange.DepartmentId = newDepartment.DepartmentId;
+                        context.Update(employeeToChange);
+                    }
+
+                }
+                await context.SaveChangesAsync();
+                return RedirectToAction("Index");
+
+            }
+
+
+            var model = new SingleDepartment();
+
+            return RedirectToAction("Index", new RouteValueDictionary(

```

Comment:
since we're redirecting in the same controller you can just write `return RedirectToAction("Index")`, though don't you want to redirect to the a Detail view for the newly created department?